### PR TITLE
Orca: rm unnecessary peripheral list in tf_runner

### PR DESCRIPTION
**Pull request by hamham223**
## Description

https://github.com/intel-analytics/BigDL/issues/7257

### 1. Why the change?

For tensorflow ray backend, should return a dict, not a list of dict.

### 2. Summary of the change

Two parts:
+ source codes: remove extra "[]" and unnecessary "itertools"
+ testing codes: add corresponding assertions to verify the return type

### 3. How to test?
- [ ] N/A
- [x] Unit test
- [ ] Application test
- [ ] Document test
- [ ] ...

Example codes to run : `python/orca/tutorial/NCF/tf_train_xshards.py`

### CODE REVIEW

File: python/orca/test/bigdl/orca/learn/ray/tf/test_tf2estimator_ray_dataset.py [hunk 1]
```diff
@@ -109,6 +109,9 @@ def test_train_and_evaluate(self):
                                                  label_cols="x", feature_cols=["y"],
                                                  data_config=data_config_args)
 
+        assert type(train_stats) == dict
+        assert type(end_eval_stats) == dict
```
Comment (hkvision) [inline python/orca/test/bigdl/orca/learn/ray/tf/test_tf2estimator_ray_dataset.py:113]: use isinstanceof

File: python/orca/src/bigdl/orca/learn/tf2/ray_estimator.py [hunk 2]
```diff
@@ -302,15 +301,15 @@ def fit(self,
...
         else:
             params["data_creator"] = data
             params["validation_data_creator"] = validation_data
             params_list = [params] * self.num_workers
 
             worker_stats = ray.get([self.remote_workers[i].step.remote(**params_list[i])
                                     for i in range(self.num_workers)])
-            worker_stats = list(itertools.chain.from_iterable(worker_stats))
+        # Tensorflow automatically synchronizes results on all the wokers
+        # thus only need to save the first one
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/learn/tf2/ray_estimator.py:312]: Tensorflow -> TensorFlow
save the first return -> return the result of the first worker

File: python/orca/test/bigdl/orca/learn/ray/tf/test_tf2estimator_ray_dataset.py [hunk 3]
```diff
@@ -109,6 +109,9 @@ def test_train_and_evaluate(self):
...
+
```
Comment (hkvision) [inline python/orca/test/bigdl/orca/learn/ray/tf/test_tf2estimator_ray_dataset.py:114]: Also need to add assert for other input types (e.g. Spark DataFrames, SparkXShards, etc.)

File: python/orca/test/bigdl/orca/learn/ray/tf/test_tf2estimator_ray_dataset.py [hunk 4]
```diff
@@ -109,6 +109,9 @@ def test_train_and_evaluate(self):
                                                  label_cols="x", feature_cols=["y"],
                                                  data_config=data_config_args)
 
+        assert isinstance(train_stats, dict)
+        assert isinstance(end_eval_stats, dict)
```
Comment (hkvision) [inline python/orca/test/bigdl/orca/learn/ray/tf/test_tf2estimator_ray_dataset.py:113]: add some error for assert
e.g.
```
assert isinstance(train_stats, dict), "fit should return a dict"
```

File: python/orca/src/bigdl/orca/learn/tf2/ray_estimator.py [hunk 5]
```diff
@@ -302,15 +301,15 @@ def fit(self,
...
         else:
             params["data_creator"] = data
             params["validation_data_creator"] = validation_data
             params_list = [params] * self.num_workers
 
             worker_stats = ray.get([self.remote_workers[i].step.remote(**params_list[i])
                                     for i in range(self.num_workers)])
-            worker_stats = list(itertools.chain.from_iterable(worker_stats))
+        # TensorFlow automatically synchronizes results on all the workers
+        # thus only need to return the result of the first worker
```
Comment (hkvision) [inline python/orca/src/bigdl/orca/learn/tf2/ray_estimator.py:312]: `and` thus

File: python/orca/test/bigdl/orca/learn/ray/tf/test_tf2estimator_ray_dataset.py (hunk 4 above)
Comment (hamham223) [reply python/orca/test/bigdl/orca/learn/ray/tf/test_tf2estimator_ray_dataset.py]: updated.

**Review by hkvision — APPROVED** [review]

**Review by sgwhat — APPROVED** [review]

**Comment by hkvision** [comment]
Two test failures are due to network. Merge this PR first.

**Pull request merged**